extensions/rvadapter/src/test/java/com/google/android/agera/rvadapter/RepositoryAdapterTest.java

```diff
@@ -201,6 +201,36 @@ public void shouldCallRecycleForOnViewRecycled() {
   }
 
   @Test
+  public void shouldCallRecycleForOnRebindViewHolder() {
+    when(repositoryPresenter.getItemCount(ALTERNATIVE_REPOSITORY_ITEM)).thenReturn(1);
+    repositoryAdapter.getItemCount(); //Trigger a refresh
+
+    repositoryAdapter.startObserving();
+    repository.accept(ALTERNATIVE_REPOSITORY_ITEM);
+    runUiThreadTasksIncludingDelayedTasks();
+    repositoryAdapter.stopObserving();
+    repositoryAdapter.onBindViewHolder(viewHolder, 0);
+    repositoryAdapter.onBindViewHolder(viewHolder, 1);
```

Comment:
@maxtroy You do have a point, the problem is that when a view goes from RepositoryPresenter A to B, the method that's going to be called is on B, meaning that A does not have a chance to release eventual resources that need to be released before binding to a new presenter. In this case B have to be aware of A somehow. When the bind operation is distributed across different presenters I think the old presenter should have a chance to execute code before handing off the view to another presenter otherwise every new presenter that reuses the same view layout have to do some tracking/housekeeping depending on the existent presenters. 

Having a "preBind" hook called on the old presenter was one of the solutions we thought about, but we ended up simplifying this into the existent "recycle" because we thought this would cover most of the cases.